### PR TITLE
a11y(menu): Add menu accessibility demo page

### PR DESCRIPTION
**Pull request by amcdnl**
Demos for a11y:

- Click on the icon will trigger menu
- Display menu options with icon

Related to #6120

### CODE REVIEW

File: src/demo-app/a11y/menu/menu-a11y.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<div class="demo-menu">
+  
+  <section>
+    <h2>Click on the icon will trigger menu</h2>
```
Comment (tinayuangao) [inline src/demo-app/a11y/menu/menu-a11y.html]: Put use case in `h2`, and put descriptions in `p`?

File: src/demo-app/a11y/menu/menu-a11y.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    <button md-icon-button [mdMenuTriggerFor]="menu1">
+      <md-icon>more_vert</md-icon>
+    </button>
+  </section>
+
+  <section>
+    <h2>Display menu options with icon</h2>
+    <button md-icon-button [mdMenuTriggerFor]="menu2">
+      <md-icon>more_vert</md-icon>
```
Comment (tinayuangao) [inline src/demo-app/a11y/menu/menu-a11y.html]: add `aria-label` or `aria-labelledby` for button without text?

File: src/demo-app/a11y/menu/menu-a11y.html [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<div class="demo-menu">
+  
+  <section>
+    <h2>Click on the icon will trigger menu</h2>
+    <md-menu #menu1="mdMenu">
+      <button md-menu-item> Settings </button>
+      <button md-menu-item> Help </button>
+    </md-menu>
+
+    <button md-icon-button [mdMenuTriggerFor]="menu1">
```
Comment (tinayuangao) [inline src/demo-app/a11y/menu/menu-a11y.html]: add `aria-label` or `aria-labelledby` for button without text?

File: src/demo-app/a11y/menu/menu-a11y.html [hunk 4]
```diff
@@ -0,0 +1,54 @@
+<div class="demo-menu">
+  
+  <section>
+    <h2>Click on the icon will trigger menu</h2>
+    <md-menu #menu1="mdMenu">
+      <button md-menu-item> Settings </button>
+      <button md-menu-item> Help </button>
```
Comment (jelbourn) [inline src/demo-app/a11y/menu/menu-a11y.html]: Can happen in a follow-up PR, but we got some advice that the buttons should _do_ something as well. Easiest thing is to just show a snackbar (which will read something w/ `aria-live`) like `You clicked "Settings"`

File: src/demo-app/a11y/menu/menu-a11y.html [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      <button md-menu-item>
+        <md-icon>notifications_off</md-icon>
+        <span>Disable alerts</span>
+      </button>
+    </md-menu>
+  </section>
+
+  <section>
+    <h2>Menu with links</h2>
+    <button md-icon-button [mdMenuTriggerFor]="menu2">
```
Comment (jelbourn) [inline src/demo-app/a11y/menu/menu-a11y.html]: The button needs a label that communicates "Here's some navigation links", maybe something like "Learn more about Angular" (with items that match)

File: src/demo-app/a11y/menu/menu-a11y.html [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+      <md-icon>more_vert</md-icon>
+    </button>
+    <md-menu #menu2="mdMenu">
+      <a href="http:/.google.com" md-menu-item>
```
Comment (jelbourn) [inline src/demo-app/a11y/menu/menu-a11y.html]: Typo in `http://`

File: src/demo-app/a11y/menu/menu-a11y.html [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        Google
+      </a>
+      <a href="http://angular.io" md-menu-item>
+        Angular
+      </a>
+      <a href="http://material.angular.io" md-menu-item>
+        Material
```
Comment (jelbourn) [inline src/demo-app/a11y/menu/menu-a11y.html]: "Angular Material"

**Comment by mmalerba** [comment]
@amcdnl can you squash this into a single commit, our presubmit script doesn't like the merge commits.

**Comment by amcdnl** [comment]
@mmalerba - I created a new clean PR - https://github.com/angular/material2/pull/6445 - its the exact same code just cleaned to one commit on latest.

**Pull request closed without merging**